tests/infer/reparam/test_haar.py

```diff
@@ -0,0 +1,85 @@
+# Copyright Contributors to the Pyro project.
```

Comment:
Tests in this file were copied nearly varbatim from `DiscreteCosineReparam` tests.